site-permissions/site-permissions-impl/src/main/java/com/duckduckgo/site/permissions/impl/SitePermissionsDialogActivityLauncher.kt

```diff
@@ -100,18 +102,24 @@ class SitePermissionsDialogActivityLauncher @Inject constructor(
         pixelParamValue: String
     ) {
         pixel.fire(SitePermissionsPixelName.SITE_PERMISSION_DIALOG_SHOWN, mapOf(PixelParameter.SITE_PERMISSION to pixelParamValue))
-        AlertDialog.Builder(activity).apply {
-            setTitle(String.format(activity.getString(titleRes), url.websiteFromGeoLocationsApiOrigin()))
-            setPositiveButton(R.string.sitePermissionsDialogAllowButton) { _, _ ->
-                pixel.fire(SitePermissionsPixelName.SITE_PERMISSION_DIALOG_ALLOWED, mapOf(PixelParameter.SITE_PERMISSION to pixelParamValue))
-                onPermissionAllowed()
-            }
-            setNegativeButton(R.string.sitePermissionsDialogDenyButton) { _, _ ->
-                pixel.fire(SitePermissionsPixelName.SITE_PERMISSION_DIALOG_DENIED, mapOf(PixelParameter.SITE_PERMISSION to pixelParamValue))
-                sitePermissionRequest.deny()
-            }
-            setCancelable(false)
-            show()
+        (activity as FragmentActivity?)?.supportFragmentManager?. let { fragmentManager ->
```

Comment:
This is a very confusing and prone to error public API now. 
For one I don't think `Activity` should be part of the public API. But even in that case, we're asking for an `Activity` but internally expect now a `FragmentActivity`.

I fail to understand why it's the responsibility of a library to actually show the dialog using `supportFragmentManager`. This is logic that belongs to the user of the library, not the library itself.

Reply:
`supportFragmentManager` was not needed at the time of the API implementation. I agree cast activity to `FragmentActivity` is not safe enough for the API. Alternative solutions I can think of:

-  I can cast it to `DuckDuckGoActivity` which should always be true if you call it from another activity ✅ 
- Pass `fragmentManager` as parameter to the API function
- Change the new component builder to something easier and default several values -> Not scoped for this PR

In the beginning I was just injecting activity context for showing dialogs, however I also needed `activity` for checking `shouldShowRequestPermissionRationale` in system permissions helper. That's why the activity is needed.

This library has wrapped all the functionality related to site permissions and system permissions combined. So you send the request and this library will handle check which permissions needs to request to the user and in case user denies/allows site permissions + system permissions it will automatically grant or deny them to the website. Therefore, the user of the library will not need to think about which dialogs or system permissions should show to the user.